ykrt/src/compile/jitc_yk/jit_ir/well_formed.rs

```diff
@@ -13,20 +15,51 @@
 //!   * [super::ICmpInst]s left and right hand side operands have the same [Ty]s.
 //!   * [Const::Int]s cannot use more bits than the corresponding [Ty::Integer] type.
 
-use super::{BinOpInst, Const, GuardInst, Inst, Module, Operand, Ty};
+use super::{BinOp, BinOpInst, Const, GuardInst, Inst, Module, Operand, Ty};
 
 impl Module {
     pub(crate) fn assert_well_formed(&self) {
         for iidx in self.iter_skipping_inst_idxs() {
             let inst = self.inst(iidx);
             match inst {
-                Inst::BinOp(BinOpInst { lhs, binop: _, rhs }) => {
-                    if lhs.unpack(self).tyidx(self) != rhs.unpack(self).tyidx(self) {
+                Inst::BinOp(BinOpInst { lhs, binop, rhs }) => {
+                    let lhs_tyidx = lhs.unpack(self).tyidx(self);
+                    if lhs_tyidx != rhs.unpack(self).tyidx(self) {
                         panic!(
                             "Instruction at position {iidx} has different types on lhs and rhs\n  {}",
                             self.inst(iidx).display(iidx, self)
                         );
                     }
+                    match binop {
+                        BinOp::Add
```

Comment:
A comment (not for this PR!) but maybe we should split into `IntBinOp` and `FloatBinOp` in the future.

Reply:
Possibly yeah. Let's see how we get on for a bit.